packages/mui-utils/src/useIsFocusVisible.ts

```diff
@@ -125,6 +125,14 @@ export interface UseIsFocusVisibleResult {
 export default function useIsFocusVisible(): UseIsFocusVisibleResult {
   const ref = React.useCallback((node) => {
     if (node != null) {
+      if (process.env.NODE_ENV !== 'production') {
+        if (!(node instanceof Element)) {
+          throw new Error(
+            'MUI: The `children` component passed to this MUI component is not an HTML or SVG element. ' +
```

Comment:
This error message applies only to Tooltip - I'd rather move it there.
Other components that use `useIsFocusVisible` may not have the `children` prop or do not call `useIsFocusVisible` on their children.

Also, let's rephrase it to say "Class components are not supported in the Tooltip. Use a function component with ref forwarded to its root DOM node."

Reply:
Apologies for the delayed reply.

It's my understanding that the error message does not apply only to Tooltip. For example, I can generate a similar error message using the Button and Link components:

https://codesandbox.io/s/friendly-snow-bk5out?file=/src/App.tsx

I haven't traced out all of the `useIsFocusVisible` calls to see what other components, if any, might be affected.

(This is why, in #33325, I wondered if updated documentation on this class component caveat should go in https://mui.com/material-ui/guides/composition/#component-prop instead of, or in addition to, the docs on Tooltip.)

Reply:
I don't think that having a warning message mentioning `component` in `useIsFocusVisible` is a good idea as it can be used in many different contexts. Let's focus on having the warning in the Tooltip now.